### PR TITLE
Starter monitoring endpoint

### PR DESCRIPTION
**Pull request by pintermate225**
I'm working with the help of this article https://howtodoinjava.com/spring-security/spring-boot-role-based-authorization/

### CODE REVIEW

File: DSN/src/main/resources/application.properties
Comment (bela-riboczki-ext) [inline DSN/src/main/resources/application.properties:1]: Please do not commit the change of the default configuration. Only do that if you remove or add something to the default configuration or we absolutely need to do so. You can ignore this file changes locally, if you want to avoid accidental commits.

File: DSN/monitoring/prometheus/prometheus.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - targets: ['localhost:9090']
+
+  - job_name: 'spring-actuator'
+    metrics_path: '/actuator/prometheus'
+    scrape_interval: 2s
+    static_configs:
+      - targets: ['localhost:8080']
+        labels:
+          group: 'production'
+    basic_auth:
```
Comment (bela-riboczki-ext) [inline DSN/monitoring/prometheus/prometheus.yml:34]: Can we use something else for authentication which is not required to commit the actual username and password?

**Review by bela-riboczki-ext — COMMENTED** [review]
Please check the comments. 

As we talked about, could you please also check what happens with the backend, when the Prometheus is not started?

File: DSN/monitoring/prometheus/prometheus.yml (hunk 1 above)
Comment (pintermate225) [reply DSN/monitoring/prometheus/prometheus.yml:34]: Yes we can, with JWT, I'm started to work on that

File: DSN/src/main/resources/application.properties
Comment (pintermate225) [reply DSN/src/main/resources/application.properties:1]: As I see, the app should work, but the localhost:9090/graph won't be accessible

Comment (bela-riboczki-ext) [reply DSN/src/main/resources/application.properties:1]: So as you expected. Thanks for checking it and letting me know.

File: DSN/src/main/resources/application.properties [hunk 2]
```diff
@@ -9,9 +9,15 @@ spring.h2.console.enabled=true
 #spring.datasource.url=jdbc:mysql://localhost:3306/social_network?serverTimezone=UTC
 #spring.datasource.username=root
 #spring.datasource.password=password
-#spring.jpa.hibernate.ddl-auto=update
```
Comment (bela-riboczki-ext) [inline DSN/src/main/resources/application.properties]: Could you please revert back the changes which is not for the prometheus endpoing?

File: DSN/pom.xml
Comment (bela-riboczki-ext) [inline DSN/pom.xml:1]: Are these package changes are necessary for the grafana and prometheus setup?
If not, could you please review them?

File: DSN/monitoring/prometheus/prometheus.yml (hunk 1 above)
Comment (bela-riboczki-ext) [reply DSN/monitoring/prometheus/prometheus.yml:34]: This looking good. You can resolve this conversation.

**Review by bela-riboczki-ext — COMMENTED** [review]
Please review my new comments.

File: DSN/backend/src/main/resources/application.properties
Comment (bela-riboczki-ext) [inline DSN/backend/src/main/resources/application.properties:1]: This file is still contains unnecessary changes about the database setup.

Comment (pintermate225) [reply DSN/backend/src/main/resources/application.properties:1]: Please check if it's fine now

File: DSN/target/generated-sources/annotations/DsnApplication.java
Comment (patvoj) [inline DSN/target/generated-sources/annotations/DsnApplication.java:1]: Why this change happened? Isn't a mistake?

Comment (pintermate225) [reply DSN/target/generated-sources/annotations/DsnApplication.java:1]: It is a mistake, I think now it is in it's proper folder

Comment (patvoj) [reply DSN/target/generated-sources/annotations/DsnApplication.java:1]: I checked it but DsnApplication is still missing in this PR :/

**Review by bela-riboczki-ext — APPROVED** [review]
Looks good from my side.

**Review by patvoj — APPROVED** [review]
All my comments were resolved, so all's good from my side. Good job!

File: docs/monitoringREADME.txt
Comment (bela-riboczki-ext) [inline docs/monitoringREADME.txt:1]: Can you make it to as a Markdown document? And please rename the file just simply **monitoring**.
So in the end you will have the document as **monitoring.md**.
